main.go

```diff
@@ -0,0 +1,260 @@
+		log.Errorf("failed to collect status from passenger: %s", err)
+		return
+	}
+	ch <- prometheus.MustNewConstMetric(e.up, prometheus.GaugeValue, 1)
+
+	// TODO: Doesn't seem to be a queryable uptime on the main process.
+	// Talk with @grobie about how to handle this.
+
+	ch <- prometheus.MustNewConstMetric(e.version, prometheus.GaugeValue, 1, info.PassengerVersion)
+
+	// This value should be 0. Not sure if I should use a gauge or a
+	// counter, as we shouldn't be counting anything.
+	ch <- prometheus.MustNewConstMetric(e.toplevelQueue, prometheus.CounterValue, parseFloat(info.TopLevelRequestsInQueue))
+	ch <- prometheus.MustNewConstMetric(e.maxProcessCount, prometheus.GaugeValue, parseFloat(info.MaxProcessCount))
+	ch <- prometheus.MustNewConstMetric(e.currentProcessCount, prometheus.GaugeValue, parseFloat(info.CurrentProcessCount))
+	ch <- prometheus.MustNewConstMetric(e.appCount, prometheus.GaugeValue, parseFloat(info.AppCount))
+
+	for _, sg := range info.SuperGroups {
+		ch <- prometheus.MustNewConstMetric(e.appQueue, prometheus.GaugeValue, parseFloat(sg.RequestsInQueue), sg.Name)
+		ch <- prometheus.MustNewConstMetric(e.appProcsSpawning, prometheus.GaugeValue, parseFloat(sg.RequestsInQueue), sg.Group.ProcessesSpawning, sg.Name)
+
+		for _, proc := range sg.Group.Processes {
+			ch <- prometheus.MustNewConstMetric(e.procMemory, prometheus.GaugeValue, parseFloat(proc.RealMemory), sg.Name, proc.PID)
+			ch <- prometheus.MustNewConstMetric(e.requestsProcessed, prometheus.CounterValue, parseFloat(proc.RequestsProcessed), sg.Name, proc.PID)
+			ch <- prometheus.MustNewConstMetric(e.procUptime, prometheus.CounterValue, parseFloat(proc.Uptime), sg.Name, proc.PID)
+
+			// Is this one really necessary?
+			ch <- prometheus.MustNewConstMetric(
+				e.procStatus, prometheus.CounterValue, 1,
+				sg.Name, proc.PID, proc.CodeRevision, proc.LifeStatus, proc.Enabled,
+			)
+		}
+	}
+
+}
+
+func (e *Exporter) status() (*Info, error) {
+	var (
+		out bytes.Buffer
+		cmd = exec.Command(e.binPath, e.args...)
+	)
+	cmd.Stdout = &out
+
+	err := cmd.Run()
+	if err != nil {
+		return nil, err
+	}
+
+	return parseOutput(&out)
+}
+
+func (e *Exporter) Describe(ch chan<- *prometheus.Desc) {
+	ch <- e.up
+	ch <- e.uptime
+	ch <- e.version
+	ch <- e.toplevelQueue
+	ch <- e.maxProcessCount
+	ch <- e.currentProcessCount
+	ch <- e.appCount
+	ch <- e.appQueue
+	ch <- e.appProcsSpawning
+	ch <- e.requestsProcessed
+	ch <- e.procUptime
+	ch <- e.procMemory
+	ch <- e.procStatus
+}
+
+func parseOutput(r io.Reader) (*Info, error) {
+	var info Info
+	err := xml.NewDecoder(r).Decode(&info)
+	if err != nil {
+		return nil, err
+	}
+	return &info, nil
+}
+
+func parseFloat(val string) float64 {
+	v, err := strconv.ParseFloat(val, 64)
+	if err != nil {
+		log.Errorf("Failed to parse %s: %v", val, err)
+		v = math.NaN()
+	}
+	return v
+}
+
+func main() {
+	var (
+		cmd           = flag.String("passenger.command", "", "Passenger command for querying passenger status.")
```

Comment:
A good default would be `/usr/sbin/passenger-status --show=xml`. Based on privileges it might not work, however, but privileges seem like something we should allow the users to handle.

https://www.phusionpassenger.com/library/admin/nginx/overall_status_report.html#to-sudo-or-not-to-sudo